lisa/feature.py

```diff
@@ -87,6 +87,16 @@ def create_setting(
         """
         return None
 
+    @classmethod
+    def create_image_requirement(
```

Comment:
The input should be `ImageSchema` not a free format dict.

Reply:
* Is there a reason `AzureImageSchema` does not inherit from Image schema?
* Are you fine with `ImageSchema` containing a free format dict field?

I don't think there is a way to completely get around needing a dict. It is similar to the `raw_capabilities` of `create_setting` where azure returns a bunch of key value pairs and the feature needs to implement the logic of how to interpret the raw capabilities.

Reply:
1. `AzureImageSchema` should inherit from `ImageSchema`
2. It doesn't need to have a free format dict in `constructors` of `ImageSchema` and its subsclasses.

There should be a static method like `create_from_raw_info`. For example, the class `AzureMarketplaceImageSchema` inherits from `AzureImageSchema`, you can parse the raw dict in the `create_from_raw_info` to generate an `AzureMarketplaceImageSchema` object.

Reply:
But a lot of the parsing needs to happen on the platform level because you need the compute client and other platform level information like which features are supported.

Also, how can I store all the feature data in the AzureMarketplaceImageSchema? Do I add a FeatureSpace to AzureImageSchema? It needs to be capable of storing information for every single Feature.

Reply:
The flow is like,

1. The platform fetches data from Azure SDK for Image.
2. Platform calls the corresponding `ImageSchema` subclass and pass in the raw image information. It supposes have all information, which needs to create features.
3. With above step, a structured `ImageSchema` is created. Call `create_settings_from_image` of each possible image feature to create settings. This logic is shared for all kinds of Azure image (market place, gallery image, community image, vhd).
4. Merge the image features with node capabilities, so it's the real capability, and ready to merge with requirements from test case and platform.

Reply:
The current implementation does not generate image capabilities, it generates image requirements which are merged with platform requirements, not platform capabilities.

You previously recommended that I modify the existing `_add_image_features` for my implementation, but this function is not positioned to perform the flow you are now suggesting. It sounds like you are now suggesting a flow which allows LISA to select an image based on other requirements; this is a much bigger ask then what I have currently implemented.

I think we should have a meeting to discuss how to move forward with this.